build/commands/lib/config.js

```diff
@@ -90,6 +90,7 @@ const Config = function () {
   this.gClientVerbose = getNPMConfig(['gclient_verbose']) || false
   this.targetArch = getNPMConfig(['target_arch']) || 'x64'
   this.targetOS = getNPMConfig(['target_os'])
+  this.targetEnvironment = getNPMConfig(['target_environment']) || ''
```

Comment:
it's best to avoid overwriting the default in gn so it should default to null

Reply:
Fixed, only setting `args.target_environment` now if `targetEnvironment` is defined